### PR TITLE
Add doughnut chart for contribution composition

### PR DESCRIPTION
**Pull request by skddc**
Enables contributors to see at a glance if all areas of a project are tended to.

screenshot from 2018-07-15 12-01-07

### CODE REVIEW

File: app/components/chart-contributions-by-type/component.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  chartData: computed('contributions', function() {
+    let kredits = this.get('contributions')
+                      .map(c => {
+                        return { kind: c.kind, amount: c.amount }
+                      }).reduce(function (kinds, c) {
+                        if (c.kind in kinds) {
+                          kinds[c.kind] = kinds[c.kind] + c.amount
+                        } else {
+                          kinds[c.kind] = c.amount;
+                        }
```
Comment (galfert) [inline app/components/chart-contributions-by-type/component.js]: I think the `if/else` block can be simplified to `kinds[c.kind] = (kinds[c.kind] || 0) + c.amount;`.

Comment (skddc) [reply app/components/chart-contributions-by-type/component.js]: Ooh, that's a good idea!

**Comment by bumi** [comment]
magic ember! :) 
looks good to me! pretty cool!

**Comment by skddc** [comment]
Thanks, everyone.

**Pull request merged**